module/VuFind/src/VuFind/Recommend/LibGuidesProfile.php

```diff
@@ -0,0 +1,241 @@
+            $client,
+            $config->General->api_base_url,
+            $config->General->client_id,
+            $config->General->client_secret
+        );
+    }
+
+    /**
+     * Store the configuration of the recommendation module.
+     *
+     * @param string $settings Settings from searches.ini.
+     *
+     * @return void
+     */
+    public function setConfig($settings)
+    {
+        // No action needed.
+    }
+
+    /**
+     * Called before the Search Results object performs its main search
+     * (specifically, in response to \VuFind\Search\SearchRunner::EVENT_CONFIGURED).
+     * This method is responsible for setting search parameters needed by the
+     * recommendation module and for reading any existing search parameters that may
+     * be needed.
+     *
+     * @param \VuFind\Search\Base\Params $params  Search parameter object
+     * @param \Laminas\Stdlib\Parameters $request Parameter object representing user
+     * request.
+     *
+     * @return void
+     *
+     * @SuppressWarnings(PHPMD.UnusedFormalParameter)
+     */
+    public function init($params, $request)
+    {
+        // No action needed.
+    }
+
+    /**
+     * Called after the Search Results object has performed its main search.  This
+     * may be used to extract necessary information from the Search Results object
+     * or to perform completely unrelated processing.
+     *
+     * @param \VuFind\Search\Base\Results $results Search results object
+     *
+     * @return void
+     */
+    public function process($results)
+    {
+        $this->results = $results;
+    }
+
+    /**
+     * Get terms related to the query.
+     *
+     * @return array
+     */
+    public function getResults()
+    {
+        // TODO: cache data, check it for staleness
+        $this->refreshData();
+
+        $query = $this->results->getParams()->getQuery();
+        $account = $this->findBestMatch($query);
+        return $account;
+    }
+
+    /**
+     * Find the VuFind account whose profile best matches the
+     * given query.
+     *
+     * @param \VuFindSearch\Query\QueryInterface $query Current search query
+     *
+     * @return array LibGuides account
+     */
+    protected function findBestMatch(\VuFindSearch\Query\QueryInterface $query)
+    {
+        $queryString = $query->getAllTerms();
+        if (!$queryString) {
+            return false;
+        }
+        $queryString = strtolower($queryString);
+
+        // Find the closest levenshtein match.
+        $minDistance = PHP_INT_MAX;
+        $subjects = array_keys($this->subjectToId);
+        $id = null;
+        foreach ($subjects as $subject) {
+            $distance = levenshtein($subject, $queryString);
+            if ($distance < $minDistance) {
+                $id = $this->subjectToId[$subject];
+                $minDistance = $distance;
+            }
+        }
+        if ($id == null) {
+            return false;
+        }
+
+        // // Find an exact match
```

Comment:
So, this could be an alternative algorithm used via configuration?  But it's a really bad algorithm as it will almost never result in a match.

Reply:
I like that idea in theory, but each algorithm except the last then has to be fail-able, which levenshtein isn't.  I do think levenshtein would be a good final-case algorithm if other, future, better ones fail -- such as the call-number based enhancement.

In any case I don't think this particular exact-match algorithm has any benefit over levenshtein *except* being fail-able, which it does 99% of the time.  So inclined to delete it.

Reply:
I've deleted it.